src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
 
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if ((key == null && current.key == null)
+                        || (key != null && key.equals(current.key))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+
+        size++;
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> current = table[index];
+
+        while (current != null) {
+            if ((key == null && current.key == null)
+                    || (key != null && key.equals(current.key))) {
+                return current.value;
+            }
+            current = current.next;
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : (key.hashCode() & 0x7FFFFFFF) % table.length;
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        threshold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
+
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                int index = (node.key == null) ? 0 :
+                        (node.key.hashCode() & 0x7FFFFFFF) % newCapacity;
+                Node<K, V> next = node.next;
+
+                node.next = newTable[index];
+                newTable[index] = node;
+
+                node = next;
+            }
+        }
+
+        table = newTable;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
     }
```

Comment:
Consider moving the `Node` class to the end of the file after all public methods. This will improve readability and maintainability by keeping the main logic of the `MyHashMap` class at the forefront.